src/drawers/drawer.ts

```diff
@@ -74,7 +74,14 @@ export class Drawer {
    * @param{any[]} data The data to be drawn
    */
   private _bindSelectionData(data: any[]) {
-    let dataElements = this.selection().data(data);
+    // if the dataset has a key, use it when binding the data   
+    let dataElements: d3.selection.Update<any>;
+    if (this._dataset && this._dataset.keyFunction()) {
```

Comment:
Since it seems that `this._dataset.keyFunction()` will never be `null` (it is by default using the index as the key, and there is a guard surrounding setting it to `null`), do we need the above guard?